### PR TITLE
Pv facility unified format

### PR DESCRIPTION
**Pull request by patrickvinograd**
Adds cemetery data alongside health facility data to single endpoint `v0/facilities/va`, with lookup by ID or bbox-bounded query. 

Right now if querying across all facility types, the backend GIS queries are serialized. I don't expect radically different behavior from them since they are all the same arcgis online server, but it would still be good to make them independent of one another. But I'd like to get this in, then iterate once we have the third data service available. 

Curious what opinions are about the "adapter" classes under models. It was clearly no good to have all the conversion happening in `models/va_facility.rb`, but is there a better pattern to follow here either for the class implementation itself or where the classes should be located in the file tree?

### CODE REVIEW

**Comment by bshyong** [comment]
@patrickvinograd from a client/front-end perspective this LGTM. I'm going to try and hook up to this API tomorrow (Tues) now that I have a PIV-enabled laptop

**Comment by patrickvinograd** [comment]
@bshyong There's not need to be on the VA VPN for this iteration - if you want to pull down this branch and run it, the arcgis endpoints are publicly available.

File: .travis.yml [hunk 1]
```diff
@@ -26,8 +26,10 @@ before_script:
   - "export EVSS_SAMPLE_CLAIMANT_USER='{\"uuid\": \"1234\", \"first_name\": \"Jane\",\"last_name\": \"Doe\",\"edipi\": \"1105051936\",\"participant_id\": \"123456789\"}'"
   - export MVI_URL='http://ps-dev.commserv.healthevet.va.gov:8110/psim_webservice/IdMWebService'
   - export EVSS_S3_UPLOADS=false
-  - export VHA_MAPSERVER_URL='https://maps.va.gov/server/rest/services/PROJECTS/Facility_Locator/MapServer'
+  - export VHA_MAPSERVER_URL='https://services3.arcgis.com/aqgBd3l68G8hEFFE/ArcGIS/rest/services/VHA_Facilities/FeatureServer'
   - export VHA_MAPSERVER_LAYER='0'
+  - export NCA_MAPSERVER_URL='https://services3.arcgis.com/aqgBd3l68G8hEFFE/ArcGIS/rest/services/NCA_Facilities/FeatureServer'
```
Comment (saneshark) [inline .travis.yml]: This is a nitpick and possibly annoying fix because it would require devops changes as well most likely, but how about just putting the host in an environment variable and then building the rest of the URI from within the class that requires it?

File: app/controllers/v0/facilities/va_controller.rb [hunk 2]
```diff
@@ -8,28 +8,32 @@ class V0::Facilities::VaController < FacilitiesController
...
-      raise ArgumentError unless params[:bbox].length == 4
-      params[:bbox].each { |x| Float(x) }
-    rescue ArgumentError
-      raise Common::Exceptions::InvalidFieldValue.new('bbox', params[:bbox])
+    super
+    if params[:type].nil? && !params[:services].nil?
+      raise Common::Exceptions::ParameterMissing.new('type', detail: TYPE_SERVICE_ERR)
+    end
+    unless params[:type].nil?
+      unless VAFacility::TYPES.include?(params[:type])
```
Comment (saneshark) [inline app/controllers/v0/facilities/va_controller.rb]: this nesting of conditions is really hard to follow. I'm thinking this is a code smell of some sort and could benefit from a) better abstraction, b) private helper methods that are more descriptive of intent.

File: app/models/vha_facility_adapter.rb [hunk 3]
```diff
@@ -1,69 +1,69 @@
...
+    m[:address] = {}
+    m[:address][:physical] = from_gis_attrs(ADDR_KEYMAP, attrs)
+    m[:address][:physical][:zip] = attrs['Zip']
+    m[:address][:physical][:zip] << '-' + attrs['Zip4'] unless attrs['Zip4'].strip.empty?
+    m[:address][:mailing] = {}
+    m[:phone] = from_gis_attrs(PHONE_KEYMAP, attrs)
+    m[:hours] = from_gis_attrs(HOURS_KEYMAP, attrs)
+    m[:services] = services_from_gis(attrs)
+    VAFacility.new(m)
+  end
```
Comment (saneshark) [inline app/models/vha_facility_adapter.rb]: Is it possibly for any of these values to be nil raising exceptions?

File: app/models/vha_facility_adapter.rb [hunk 4]
```diff
@@ -95,7 +95,13 @@ def self.from_gis_attrs(km, attrs)
     'WellnessAndPreventativeCare' => []
   }.freeze
 
-  def self.services_from_gis(attrs)
+  def from_gis_attrs(km, attrs)
```
Comment (saneshark) [inline app/models/vha_facility_adapter.rb]: a comment here to inform intent would be helpful

File: app/models/vha_facility_adapter.rb [hunk 5]
```diff
@@ -95,7 +95,13 @@ def self.from_gis_attrs(km, attrs)
...
+    km.each_with_object({}) do |(k, v), h|
+      h[k] = (attrs[v].respond_to?(:strip) ? attrs[v].strip : attrs[v])
+    end
+  end
+
+  def services_from_gis(attrs)
```
Comment (saneshark) [inline app/models/vha_facility_adapter.rb]: similarly i think a comment here would be helpful

File: config/application.yml.example [hunk 6]
```diff
@@ -17,8 +17,10 @@ MHV_SM_USER_ID: '10616687'
 EVSS_BASE_URL: 'https://test.vets.gov'
 EVSS_SAMPLE_CLAIMANT_USER: '{"uuid": "1234", "first_name": "Jane","last_name": "Doe","edipi": "1105051936","participant_id": "123456789"}'
 EVSS_S3_UPLOADS: 'false'
-VHA_MAPSERVER_URL: 'https://maps.va.gov/server/rest/services/PROJECTS/Facility_Locator/MapServer'
+VHA_MAPSERVER_URL: 'https://services3.arcgis.com/aqgBd3l68G8hEFFE/ArcGIS/rest/services/VHA_Facilities/FeatureServer'
 VHA_MAPSERVER_LAYER: '0'
+NCA_MAPSERVER_URL: 'https://services3.arcgis.com/aqgBd3l68G8hEFFE/ArcGIS/rest/services/NCA_Facilities/FeatureServer'
```
Comment (saneshark) [inline config/application.yml.example]: same comment as before about using ENV just for common host name

File: .travis.yml (hunk 1 above)
Comment (patrickvinograd) [inline .travis.yml]: Up until recently it was not clear whether the different services were going to share a base URL or not. I'm still not 100% credulous that e.g. the "NCA_Facilities" part of the path is going to remain stable so I'm hesitant to put it in code vs. config. I am likely to eliminate the separate "LAYER" env vars in the future by switching to a standard HTTP client instead of the geoservices gem which doesn't add much value as it's just assembly a query string.

File: app/models/vha_facility_adapter.rb (hunk 3 above)
Comment (patrickvinograd) [inline app/models/vha_facility_adapter.rb]: In the current GIS data set, empty values are represented as an empty string - I can pull down the complete facilities data and verify. But, it looks like it's possible for ArcGIS REST to return nulls. I think only the `Zip4` processing above would be thrown off by a nil, so I'll fix that up.

**Comment by saneshark** [comment]
So, maybe getting a second set of eyes on this wouldn't hurt, but LGTM

**Comment by patrickvinograd** [comment]
@bshyong already gave this a look so I'm going to prepare a devops PR for the env variables and then hit the button.
